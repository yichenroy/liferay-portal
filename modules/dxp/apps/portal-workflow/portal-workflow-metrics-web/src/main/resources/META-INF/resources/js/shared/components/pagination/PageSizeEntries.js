import autobind from 'autobind-decorator';
import Icon from '../Icon';
import PageSizeItem from './PageSizeItem';
import React from 'react';

/**
 * @class
 * @memberof shared/components
 * */
export default class PageSizeEntries extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			selectedPageSize: props.selectedPageSize
		};
	}

	@autobind
	setPageSize(pageSize) {
		const {onSelectPageSize} = this.props;

		this.setState({selectedPageSize: pageSize});
		onSelectPageSize(pageSize);
	}

	render() {
		const {pageSizeEntries} = this.props;
		const {selectedPageSize} = this.state;

		return (
			<div className="dropdown pagination-items-per-page">
				<a
					aria-expanded="false"
					aria-haspopup="true"
					className="dropdown-toggle"
					data-toggle="dropdown"
					href="#1"
					role="button"
				>
					{`${selectedPageSize} ${Liferay.Language.get('entries')}`}
					<Icon iconName="caret-double-l" />
				</a>
				<div className="dropdown-menu dropdown-menu-top">
					{pageSizeEntries.map((pageSize, index) => (
						<PageSizeItem
							key={`${index}_${pageSize}`}
							onChangePageSize={this.setPageSize}
							pageSize={pageSize}
						/>
					))}
				</div>
			</div>
		);
	}
}