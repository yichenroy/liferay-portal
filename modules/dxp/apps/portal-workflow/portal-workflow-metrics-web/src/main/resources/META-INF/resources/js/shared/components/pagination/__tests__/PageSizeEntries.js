import PageSizeEntries from '../PageSizeEntries';
import React from 'react';

test('Should render component', () => {
	const component = shallow(
		<PageSizeEntries pageSizeEntries={[10, 20, 30, 40]} selectedPageSize={10} />
	);

	expect(component).toMatchSnapshot();
});

test('Should change page size', () => {
	const onSelectPageSize = () => pageSize => pageSize;

	const component = shallow(
		<PageSizeEntries
			onSelectPageSize={onSelectPageSize()}
			pageSizeEntries={[10, 20, 30, 40]}
			selectedPageSize={30}
		/>
	);

	const instance = component.instance();

	instance.setPageSize(30);
	expect(component.state('selectedPageSize')).toBe(30);
});