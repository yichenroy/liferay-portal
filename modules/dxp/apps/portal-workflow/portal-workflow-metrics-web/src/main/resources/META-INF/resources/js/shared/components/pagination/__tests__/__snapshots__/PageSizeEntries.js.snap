// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should change page size 1`] = `
<div
  className="dropdown pagination-items-per-page"
>
  <a
    aria-expanded="false"
    aria-haspopup="true"
    className="dropdown-toggle"
    data-toggle="dropdown"
    href="#1"
    role="button"
  >
    30 Entries
    <Icon
      iconName="caret-double-l"
    />
  </a>
  <div
    className="dropdown-menu dropdown-menu-top"
  >
    <withRouter(PageSizeItem)
      key="0_10"
      pageSize={10}
    />
    <withRouter(PageSizeItem)
      key="1_20"
      pageSize={20}
    />
    <withRouter(PageSizeItem)
      key="2_30"
      pageSize={30}
    />
    <withRouter(PageSizeItem)
      key="3_40"
      pageSize={40}
    />
  </div>
</div>
`;

exports[`Should render component 1`] = `
<div
  className="dropdown pagination-items-per-page"
>
  <a
    aria-expanded="false"
    aria-haspopup="true"
    className="dropdown-toggle"
    data-toggle="dropdown"
    href="#1"
    role="button"
  >
    10 Entries
    <Icon
      iconName="caret-double-l"
    />
  </a>
  <div
    className="dropdown-menu dropdown-menu-top"
  >
    <withRouter(PageSizeItem)
      key="0_10"
      pageSize={10}
    />
    <withRouter(PageSizeItem)
      key="1_20"
      pageSize={20}
    />
    <withRouter(PageSizeItem)
      key="2_30"
      pageSize={30}
    />
    <withRouter(PageSizeItem)
      key="3_40"
      pageSize={40}
    />
  </div>
</div>
`;

exports[`Should render with default deltas 1`] = `
<MockRouter>
  <MemoryRouter
    initialEntries={
      Array [
        Object {
          "match": Object {
            "params": Object {
              "page": 1,
              "search": undefined,
              "sort": undefined,
            },
          },
          "pathname": "/processes/1/10/title%3Aasc",
          "search": "?backPath=%2F",
        },
        Object {
          "pathname": "/",
          "search": "?backPath=%2F",
        },
        Object {
          "pathname": "/slas/1234",
          "search": "?backPath=%2Fprocesses",
        },
      ]
    }
    keyLength={0}
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "match": Object {
                "params": Object {
                  "page": 1,
                  "search": undefined,
                  "sort": undefined,
                },
              },
              "pathname": "/processes/1/10/title%3Aasc",
              "search": "?backPath=%2F",
            },
            Object {
              "hash": "",
              "pathname": "/",
              "search": "?backPath=%2F",
            },
            Object {
              "hash": "",
              "pathname": "/slas/1234",
              "search": "?backPath=%2Fprocesses",
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 3,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "match": Object {
              "params": Object {
                "page": 1,
                "search": undefined,
                "sort": undefined,
              },
            },
            "pathname": "/processes/1/10/title%3Aasc",
            "search": "?backPath=%2F",
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <PageSizeEntries
        selectedPageSize={30}
      >
        <div
          className="dropdown pagination-items-per-page"
        >
          <a
            aria-expanded="false"
            aria-haspopup="true"
            className="dropdown-toggle"
            data-toggle="dropdown"
            href="#1"
            role="button"
          >
            30 Entries
            <Icon
              iconName="caret-double-l"
            >
              <svg
                className="lexicon-icon lexicon-icon-caret-double-l"
                focusable="false"
                role="presentation"
              >
                <use
                  href="http://localhost:8080/o/admin-theme/images/lexicon/icons.svg#caret-double-l"
                />
              </svg>
            </Icon>
          </a>
          <div
            className="dropdown-menu dropdown-menu-top"
          >
            <withRouter(PageSizeItem)
              key="0_5"
              pageSize={5}
            >
              <Route>
                <PageSizeItem
                  history={
                    Object {
                      "action": "POP",
                      "block": [Function],
                      "canGo": [Function],
                      "createHref": [Function],
                      "entries": Array [
                        Object {
                          "hash": "",
                          "match": Object {
                            "params": Object {
                              "page": 1,
                              "search": undefined,
                              "sort": undefined,
                            },
                          },
                          "pathname": "/processes/1/10/title%3Aasc",
                          "search": "?backPath=%2F",
                        },
                        Object {
                          "hash": "",
                          "pathname": "/",
                          "search": "?backPath=%2F",
                        },
                        Object {
                          "hash": "",
                          "pathname": "/slas/1234",
                          "search": "?backPath=%2Fprocesses",
                        },
                      ],
                      "go": [Function],
                      "goBack": [Function],
                      "goForward": [Function],
                      "index": 0,
                      "length": 3,
                      "listen": [Function],
                      "location": Object {
                        "hash": "",
                        "match": Object {
                          "params": Object {
                            "page": 1,
                            "search": undefined,
                            "sort": undefined,
                          },
                        },
                        "pathname": "/processes/1/10/title%3Aasc",
                        "search": "?backPath=%2F",
                      },
                      "push": [Function],
                      "replace": [Function],
                    }
                  }
                  location={
                    Object {
                      "hash": "",
                      "match": Object {
                        "params": Object {
                          "page": 1,
                          "search": undefined,
                          "sort": undefined,
                        },
                      },
                      "pathname": "/processes/1/10/title%3Aasc",
                      "search": "?backPath=%2F",
                    }
                  }
                  match={
                    Object {
                      "isExact": false,
                      "params": Object {},
                      "path": "/",
                      "url": "/",
                    }
                  }
                  pageSize={5}
                >
                  <Link
                    className="dropdown-item"
                    replace={false}
                    to={
                      Object {
                        "pathname": "/",
                        "search": "?backPath=%2F",
                      }
                    }
                  >
                    <a
                      className="dropdown-item"
                      href="/?backPath=%2F"
                      onClick={[Function]}
                    >
                      5
                    </a>
                  </Link>
                </PageSizeItem>
              </Route>
            </withRouter(PageSizeItem)>
            <withRouter(PageSizeItem)
              key="1_10"
              pageSize={10}
            >
              <Route>
                <PageSizeItem
                  history={
                    Object {
                      "action": "POP",
                      "block": [Function],
                      "canGo": [Function],
                      "createHref": [Function],
                      "entries": Array [
                        Object {
                          "hash": "",
                          "match": Object {
                            "params": Object {
                              "page": 1,
                              "search": undefined,
                              "sort": undefined,
                            },
                          },
                          "pathname": "/processes/1/10/title%3Aasc",
                          "search": "?backPath=%2F",
                        },
                        Object {
                          "hash": "",
                          "pathname": "/",
                          "search": "?backPath=%2F",
                        },
                        Object {
                          "hash": "",
                          "pathname": "/slas/1234",
                          "search": "?backPath=%2Fprocesses",
                        },
                      ],
                      "go": [Function],
                      "goBack": [Function],
                      "goForward": [Function],
                      "index": 0,
                      "length": 3,
                      "listen": [Function],
                      "location": Object {
                        "hash": "",
                        "match": Object {
                          "params": Object {
                            "page": 1,
                            "search": undefined,
                            "sort": undefined,
                          },
                        },
                        "pathname": "/processes/1/10/title%3Aasc",
                        "search": "?backPath=%2F",
                      },
                      "push": [Function],
                      "replace": [Function],
                    }
                  }
                  location={
                    Object {
                      "hash": "",
                      "match": Object {
                        "params": Object {
                          "page": 1,
                          "search": undefined,
                          "sort": undefined,
                        },
                      },
                      "pathname": "/processes/1/10/title%3Aasc",
                      "search": "?backPath=%2F",
                    }
                  }
                  match={
                    Object {
                      "isExact": false,
                      "params": Object {},
                      "path": "/",
                      "url": "/",
                    }
                  }
                  pageSize={10}
                >
                  <Link
                    className="dropdown-item"
                    replace={false}
                    to={
                      Object {
                        "pathname": "/",
                        "search": "?backPath=%2F",
                      }
                    }
                  >
                    <a
                      className="dropdown-item"
                      href="/?backPath=%2F"
                      onClick={[Function]}
                    >
                      10
                    </a>
                  </Link>
                </PageSizeItem>
              </Route>
            </withRouter(PageSizeItem)>
            <withRouter(PageSizeItem)
              key="2_20"
              pageSize={20}
            >
              <Route>
                <PageSizeItem
                  history={
                    Object {
                      "action": "POP",
                      "block": [Function],
                      "canGo": [Function],
                      "createHref": [Function],
                      "entries": Array [
                        Object {
                          "hash": "",
                          "match": Object {
                            "params": Object {
                              "page": 1,
                              "search": undefined,
                              "sort": undefined,
                            },
                          },
                          "pathname": "/processes/1/10/title%3Aasc",
                          "search": "?backPath=%2F",
                        },
                        Object {
                          "hash": "",
                          "pathname": "/",
                          "search": "?backPath=%2F",
                        },
                        Object {
                          "hash": "",
                          "pathname": "/slas/1234",
                          "search": "?backPath=%2Fprocesses",
                        },
                      ],
                      "go": [Function],
                      "goBack": [Function],
                      "goForward": [Function],
                      "index": 0,
                      "length": 3,
                      "listen": [Function],
                      "location": Object {
                        "hash": "",
                        "match": Object {
                          "params": Object {
                            "page": 1,
                            "search": undefined,
                            "sort": undefined,
                          },
                        },
                        "pathname": "/processes/1/10/title%3Aasc",
                        "search": "?backPath=%2F",
                      },
                      "push": [Function],
                      "replace": [Function],
                    }
                  }
                  location={
                    Object {
                      "hash": "",
                      "match": Object {
                        "params": Object {
                          "page": 1,
                          "search": undefined,
                          "sort": undefined,
                        },
                      },
                      "pathname": "/processes/1/10/title%3Aasc",
                      "search": "?backPath=%2F",
                    }
                  }
                  match={
                    Object {
                      "isExact": false,
                      "params": Object {},
                      "path": "/",
                      "url": "/",
                    }
                  }
                  pageSize={20}
                >
                  <Link
                    className="dropdown-item"
                    replace={false}
                    to={
                      Object {
                        "pathname": "/",
                        "search": "?backPath=%2F",
                      }
                    }
                  >
                    <a
                      className="dropdown-item"
                      href="/?backPath=%2F"
                      onClick={[Function]}
                    >
                      20
                    </a>
                  </Link>
                </PageSizeItem>
              </Route>
            </withRouter(PageSizeItem)>
            <withRouter(PageSizeItem)
              key="3_30"
              pageSize={30}
            >
              <Route>
                <PageSizeItem
                  history={
                    Object {
                      "action": "POP",
                      "block": [Function],
                      "canGo": [Function],
                      "createHref": [Function],
                      "entries": Array [
                        Object {
                          "hash": "",
                          "match": Object {
                            "params": Object {
                              "page": 1,
                              "search": undefined,
                              "sort": undefined,
                            },
                          },
                          "pathname": "/processes/1/10/title%3Aasc",
                          "search": "?backPath=%2F",
                        },
                        Object {
                          "hash": "",
                          "pathname": "/",
                          "search": "?backPath=%2F",
                        },
                        Object {
                          "hash": "",
                          "pathname": "/slas/1234",
                          "search": "?backPath=%2Fprocesses",
                        },
                      ],
                      "go": [Function],
                      "goBack": [Function],
                      "goForward": [Function],
                      "index": 0,
                      "length": 3,
                      "listen": [Function],
                      "location": Object {
                        "hash": "",
                        "match": Object {
                          "params": Object {
                            "page": 1,
                            "search": undefined,
                            "sort": undefined,
                          },
                        },
                        "pathname": "/processes/1/10/title%3Aasc",
                        "search": "?backPath=%2F",
                      },
                      "push": [Function],
                      "replace": [Function],
                    }
                  }
                  location={
                    Object {
                      "hash": "",
                      "match": Object {
                        "params": Object {
                          "page": 1,
                          "search": undefined,
                          "sort": undefined,
                        },
                      },
                      "pathname": "/processes/1/10/title%3Aasc",
                      "search": "?backPath=%2F",
                    }
                  }
                  match={
                    Object {
                      "isExact": false,
                      "params": Object {},
                      "path": "/",
                      "url": "/",
                    }
                  }
                  pageSize={30}
                >
                  <Link
                    className="dropdown-item"
                    replace={false}
                    to={
                      Object {
                        "pathname": "/",
                        "search": "?backPath=%2F",
                      }
                    }
                  >
                    <a
                      className="dropdown-item"
                      href="/?backPath=%2F"
                      onClick={[Function]}
                    >
                      30
                    </a>
                  </Link>
                </PageSizeItem>
              </Route>
            </withRouter(PageSizeItem)>
            <withRouter(PageSizeItem)
              key="4_50"
              pageSize={50}
            >
              <Route>
                <PageSizeItem
                  history={
                    Object {
                      "action": "POP",
                      "block": [Function],
                      "canGo": [Function],
                      "createHref": [Function],
                      "entries": Array [
                        Object {
                          "hash": "",
                          "match": Object {
                            "params": Object {
                              "page": 1,
                              "search": undefined,
                              "sort": undefined,
                            },
                          },
                          "pathname": "/processes/1/10/title%3Aasc",
                          "search": "?backPath=%2F",
                        },
                        Object {
                          "hash": "",
                          "pathname": "/",
                          "search": "?backPath=%2F",
                        },
                        Object {
                          "hash": "",
                          "pathname": "/slas/1234",
                          "search": "?backPath=%2Fprocesses",
                        },
                      ],
                      "go": [Function],
                      "goBack": [Function],
                      "goForward": [Function],
                      "index": 0,
                      "length": 3,
                      "listen": [Function],
                      "location": Object {
                        "hash": "",
                        "match": Object {
                          "params": Object {
                            "page": 1,
                            "search": undefined,
                            "sort": undefined,
                          },
                        },
                        "pathname": "/processes/1/10/title%3Aasc",
                        "search": "?backPath=%2F",
                      },
                      "push": [Function],
                      "replace": [Function],
                    }
                  }
                  location={
                    Object {
                      "hash": "",
                      "match": Object {
                        "params": Object {
                          "page": 1,
                          "search": undefined,
                          "sort": undefined,
                        },
                      },
                      "pathname": "/processes/1/10/title%3Aasc",
                      "search": "?backPath=%2F",
                    }
                  }
                  match={
                    Object {
                      "isExact": false,
                      "params": Object {},
                      "path": "/",
                      "url": "/",
                    }
                  }
                  pageSize={50}
                >
                  <Link
                    className="dropdown-item"
                    replace={false}
                    to={
                      Object {
                        "pathname": "/",
                        "search": "?backPath=%2F",
                      }
                    }
                  >
                    <a
                      className="dropdown-item"
                      href="/?backPath=%2F"
                      onClick={[Function]}
                    >
                      50
                    </a>
                  </Link>
                </PageSizeItem>
              </Route>
            </withRouter(PageSizeItem)>
            <withRouter(PageSizeItem)
              key="5_75"
              pageSize={75}
            >
              <Route>
                <PageSizeItem
                  history={
                    Object {
                      "action": "POP",
                      "block": [Function],
                      "canGo": [Function],
                      "createHref": [Function],
                      "entries": Array [
                        Object {
                          "hash": "",
                          "match": Object {
                            "params": Object {
                              "page": 1,
                              "search": undefined,
                              "sort": undefined,
                            },
                          },
                          "pathname": "/processes/1/10/title%3Aasc",
                          "search": "?backPath=%2F",
                        },
                        Object {
                          "hash": "",
                          "pathname": "/",
                          "search": "?backPath=%2F",
                        },
                        Object {
                          "hash": "",
                          "pathname": "/slas/1234",
                          "search": "?backPath=%2Fprocesses",
                        },
                      ],
                      "go": [Function],
                      "goBack": [Function],
                      "goForward": [Function],
                      "index": 0,
                      "length": 3,
                      "listen": [Function],
                      "location": Object {
                        "hash": "",
                        "match": Object {
                          "params": Object {
                            "page": 1,
                            "search": undefined,
                            "sort": undefined,
                          },
                        },
                        "pathname": "/processes/1/10/title%3Aasc",
                        "search": "?backPath=%2F",
                      },
                      "push": [Function],
                      "replace": [Function],
                    }
                  }
                  location={
                    Object {
                      "hash": "",
                      "match": Object {
                        "params": Object {
                          "page": 1,
                          "search": undefined,
                          "sort": undefined,
                        },
                      },
                      "pathname": "/processes/1/10/title%3Aasc",
                      "search": "?backPath=%2F",
                    }
                  }
                  match={
                    Object {
                      "isExact": false,
                      "params": Object {},
                      "path": "/",
                      "url": "/",
                    }
                  }
                  pageSize={75}
                >
                  <Link
                    className="dropdown-item"
                    replace={false}
                    to={
                      Object {
                        "pathname": "/",
                        "search": "?backPath=%2F",
                      }
                    }
                  >
                    <a
                      className="dropdown-item"
                      href="/?backPath=%2F"
                      onClick={[Function]}
                    >
                      75
                    </a>
                  </Link>
                </PageSizeItem>
              </Route>
            </withRouter(PageSizeItem)>
          </div>
        </div>
      </PageSizeEntries>
    </Router>
  </MemoryRouter>
</MockRouter>
`;
